libs/shared/methodologies/bold/getters/src/document.getters.ts

```diff
@@ -9,43 +10,47 @@ import {
   DocumentEventName,
   MassIdDocumentActorType,
 } from '@carrot-fndn/shared/methodologies/bold/types';
+import { getYear } from 'date-fns';
 
 import { getEventAttributeValue } from './event.getters';
 
 const { DROP_OFF, EMISSION_AND_COMPOSTING_METRICS, PICK_UP, RULES_METADATA } =
   DocumentEventName;
 const { PROCESSOR, RECYCLER, WASTE_GENERATOR } = MassIdDocumentActorType;
 
-export const getLastEmissionAndCompostingMetricsEvent = (
-  document: Document,
-): DocumentEvent | undefined => {
-  const emissionAndCompostingMetricsEvents = document.externalEvents?.filter(
-    (event) => event.name.toString().includes(EMISSION_AND_COMPOSTING_METRICS),
+interface LastYearEmissionAndCompostingMetricsEventParameters {
+  documentWithEmissionAndCompostingMetricsEvent: Document;
+  documentWithYear: Document;
+}
+
+export const getLastYearEmissionAndCompostingMetricsEvent = ({
+  documentWithEmissionAndCompostingMetricsEvent,
+  documentWithYear,
+}: LastYearEmissionAndCompostingMetricsEventParameters):
+  | DocumentEvent
+  | undefined => {
+  const currentYear = getYear(documentWithYear.externalCreatedAt);
+  const lastYear = currentYear - 1;
+
+  return documentWithEmissionAndCompostingMetricsEvent.externalEvents?.find(
+    (event) => {
+      if (event.name.toString().includes(EMISSION_AND_COMPOSTING_METRICS)) {
```

Comment:
Maybe change to startsWith?
```suggestion
      if (event.name.toString().startsWith(EMISSION_AND_COMPOSTING_METRICS)) {
```